### ISSUE TITLE
[RRFC] npm init should also run git init if it git is installed or the folder is not a git repo already

### ISSUE THREAD

**Issue by Rafi993** [npm/rfcs#411]
## Motivation ("The Why")
* Most of the time when developers do `$ npm init` in a project they usually run `$ git init` to set it as a git repo
* In other eco systems like rust in tools like cargo when you do `cargo init` checks if you have git installed if so initialises it as a git repo (it it already not a git repo) 
 * If the developer does not want to use git but use other version controls (or none) they could specify it in a flag like
    ``` $ cargo --vcs```
* We could do similar thing in could be done in npm. Even-though this is not a major change it does improve the developer experience little bit

### Example
* You go to a folder and run `$ npm init -y` it creates a `package.json` with default values and also initialises it as a git repo

#### Current Behaviour
* Right now when you do `$ npm init` it asks you a set of questions then it initialises the `package.json`
* Or when you do `$ npm init -y` it creates the package.json with default values

#### Desired Behaviour
* When you run `$ npm init` in addition to exiting questions ask to the developer the should be question for version control and git should be the default value (if they don't have git installed it should say git is not installed) and on completion it will initialise the git repo
* Similarly when they run `$ npm init -y` it will initialise the folder as git repo if it is already not a git repo and git is installed

### References

* n/a

**Comment by ljharb** [comment]
How would npm know if the dir is already in a repo or not? It might not be in a git repo - it might be mercurial, or svn, or perforce, or something invented next week.

**Comment by Rafi993** [comment]
> How would npm know if the dir is already in a repo or not? It might not be in a git repo - it might be mercurial, or svn, or perforce, or something invented next week.

We could do something similar to cargo https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/cargo_new.rs#L674-L703 by default assume the version control is `git` if other version control needs to be used you may need to specify. This means it makes things easier for majority of users but for users who use other vcs they need to specify the flag.

**Comment by ljharb** [comment]
What if it’s intended to be a subfolder in a not-yet-inited monorepo?

**Comment by Rafi993** [comment]
> What if it’s intended to be a subfolder in a not-yet-inited monorepo?

You are right it does complicate things rather than simplifying. Since cargo follows different approach to those issues replicating it might not be possible I did not think this through. Thanks for your time @ljharb. Feel free to close this RFC if this could not be taken further.
